src/plugins/dashboard/public/dashboard_app/_dashboard_app_strings.ts

```diff
@@ -154,6 +154,11 @@ export const shareModalStrings = {
       defaultMessage:
         'One or more panels on this dashboard have changed. Before you generate a snapshot, save the dashboard.',
     }),
+  getDraftShareWarning: () =>
+    i18n.translate('dashboard.snapshotShare.draftWarning', {
+      defaultMessage:
+        'One or more panels on this dashboard have changed, save the dashboard to create a permanent link.',
+    }),
```

Comment:
I didn't stick to what we had previously since it mentions snapshot and that's a concept we've sort of done away with, instead I pretty much adapted  the error message we have for lens, I expected that there was going to be some deliberation about the text, feel free to suggest a more suitable alternative. 

Also thanks for pointing out the distinction for when said error message gets displayed for users, I'd make adjustments. On wether it's necessary to display the message still when a user has short URL permission,  I'd say it's really your team's call about that. I do agree we should definitely have a more explicit warning for instances where the user doesn't have short URL permissions.

Reply:
> On wether it's necessary to display the message still when a user has short URL permission, I'd say it's really your team's call about that

I thought about this more and I think we should always show consistent warnings, regardless of short URL permissions - i.e. when the dashboard has unsaved changes **to the panels**, show a panel-specific warning (something like `"One or more panels on this dashboard have changed. Before generating a link, save the dashboard."`) and when there are other unsaved changes, a more general warning can be displayed (something like `"This dashboard has unsaved changes. Consider saving your dashboard before generating the link."`)

I am trying to emphasize to the user that having unsaved changes to panels is **much more high risk** than other unsaved changes; it results in an extremely long URL, it has risk of overwriting information, etc. I'm not really sure if there is a better way to do this in the new design.